src/DynamicData.Tests/List/MergeManyChangeSetsListFixture.cs

```diff
@@ -39,6 +46,71 @@ public MergeManyChangeSetsListFixture()
         _animalResults = _animalOwners.Connect().MergeManyChangeSets(owner => owner.Animals.Connect()).AsAggregator();
     }
 
+    [Theory]
+    [InlineData(5, 7)]
+    [InlineData(10, 50)]
+    [InlineData(10, 1_000)]
+    [InlineData(200, 500)]
+    [InlineData(1_000, 10)]
+    public async Task MultiThreadedStressTest(int ownerCount, int animalCount) =>
+        _ = await AddRemoveAnimalsStress(ownerCount, animalCount, TaskPoolScheduler.Default)
+            .Finally(CheckResultContents);
+
+    [Theory]
+    [InlineData(5, 7)]
+    [InlineData(5, 200)]
+    [InlineData(10, 100)]
+    [InlineData(20, 50)]
+    [InlineData(100, 10)]
+    public void NoDeadlockOrExceptionIfSubscribeDuringModify(int ownerCount, int animalCount)
+    {
+        // Not used so don't let it waste time
+        _animalResults.Dispose();
+
+        // Arrange
+        Func<Task> CreateTest(IScheduler sch, int owners, int animals) =>
+            async () =>
+            {
+                var mergeAnimals = _animalOwners.Connect().MergeManyChangeSets(owner => owner.Animals.Connect());
+
+                var addingAnimals = true;
+
+                using var addOwners = GenerateOwners(sch)
+                    .Take(owners)
+                    .StressAddRemove(_animalOwners, _ => GetRemoveTime(), sch)
+                    .Finally(() => _animalOwners.Dispose())
+                    .Subscribe();
+
+                using var addAnimals = _animalOwners.Connect()
+                    .MergeMany(owner => AddRemoveAnimals(owner, sch, animals))
+                    .Finally(() => addingAnimals = false)
+                    .Subscribe();
+
+                do
+                {
+                    // Ensure items are being added asynchronously before subscribing to the animal changes
+                    await Task.Yield();
+
+                    {
+                        // Subscribe
+                        var mergedSub = mergeAnimals.Subscribe();
+
+                        // Let other threads run
+                        await Task.Yield();
+
+                        // Unsubscribe
+                        mergedSub.Dispose();
+                    }
+                }
+                while (addingAnimals);
+            };
+
+        // Act
+
+        // Assert
+        CreateTest(TaskPoolScheduler.Default, ownerCount, animalCount).Should().NotThrowAsync();
```

Comment:
Are we just going for, like, clarification of intent here? Like, the reason there's no other assertions is that we're specifically testing to look for exceptions?

Reply:
Exceptions or deadlocks.  Testing the final output is in the other test.
This one is just sub/unsub repeatedly while other threads do add/remove to both the parent and child containers.

I included this test because of the exception that was reported the other day that occurred when subscribing while something was modifying the container.

Reply:
I meant specifically wrapping the test in a delegate, and then invoking it. Does that account for deadlocks? The test itself is fine, the delegate abstraction just seems pointless.

Reply:
`.Should().NotThrowAsync()` only works on `Func<Task>`.  Is there a better way to explicitly test for the lack of an exception?  I'm certainly open to suggestions.

Reply:
No, my point is that explicitly testing for the lack of an exception seems pointless to me. ALL tests test for the lack of exceptions, implicitly.

Reply:
Okay, I'll fix it.